app/models/bot.rb

```diff
@@ -11,4 +11,10 @@ def eligible_collaborators
         .order(:username)
         .map{ |u| [u.username, u.id] }
   end
+
+  def active_instance
+    BotInstance.where(bot: self)
+               .where('last_ping > ?', 3.minutes.ago)
```

Comment:
This is supposed to be the instance that *should* be active, not the ones that are currently active.  Maybe I should rename it to `preferred_instance`?